mbus-api/src/event_traits.rs

```diff
@@ -0,0 +1,55 @@
+use crate::event::{Component, EventMessage, EventMeta, EventSource, Version};
+use chrono::Utc;
+use once_cell::sync::OnceCell;
+use std::str::FromStr;
+
+/// Once cell static variable to store the component field of the event source.
+static COMPONENT: OnceCell<Component> = OnceCell::new();
+
+/// Initialize the event source component with the service name.
+pub fn initilize_source_component(comp: String) {
```

Comment:
This will be more useful as &String can be coerced into &str
```suggestion
pub fn initilize_source_component(comp: &str) {
```